community_supplied/Chassis/netconf-system-cpu-temperatures.rule

```diff
@@ -0,0 +1,147 @@
+healthbot {
+    topic chassis.temperature  {
+        rule check-netconf-chassis-temperatures {
+	    synopsis "Chassis component temperature change detector";
+            description "Collects temperature details from chassis components and notify anomalies based on threshold values ";
+		    /*
+            * Use componet-name as key for rule.
+            */
+            keys component-name;
+            function temp_degrees {
+			    description "Python application collects string from the iagent sensor and pick the temperature as integer for all chassis components";
+                path temp_degrees.py;
+                method temp_degrees;
+                argument temperature {
+                    mandatory;
+                }
+            }
+            sensor chassis-sensor {
+			    synopsis "Chassis temperature sensor initialization";
+                description "iAgent sensor to collect data from network device"; 
+                iAgent {
+                    file chassis-env-info.yml;
+                    table ChassisStatsTable;
+                    frequency 300s;
+                }
+            }
+			/*
+            * Fields defined with variables, dynamic thresholds and constant values
+            */
+            field component-name {
+                sensor chassis-sensor {
+                    path name;
+                }
+                type string;
+                description "Name of the component";
+            }
+            field dt-temperature-value {
+                formula {
+                    dynamic-threshold {
+                        algorithm 3sigma;
+                        learning-period 7d;
+                        pattern-periodicity 1d,1h;
+                        field-name "$temperature-value";
+                    }
+                }                                   
+                type integer;
+                description "dynamic threshold value for component temperature";
+            }
+            field temperature-value {
+                formula {
+                    user-defined-function {
+                        function-name temp_degrees;
+                        argument temperature "$temperature";
+                    }
+                }
+                type integer;
+            }
+            field threshold {
+                constant {
+                    value "{{threshold-value}}";
+                }
+                type integer;
+                description "Threshold value for component temperature";
+            }
+			/*
+            * Anomaly detection logic for chassis components.
+            */
+            trigger component-temperature {
+			    /*
+                * Sets color to red and sends out an anomaly notification when
+                * temperature value increases more than threshold static ($threshold) value
+                * 
+                */ 
+                frequency 300s;
+                term is-component-temperature-is-abnormal {

```

Comment:
Can you please rename this to is-component-temperature-abnormal? Or even better - has-component-temperature-breached-static-threshold?

Reply:
Modified

Reply:
Changed